internal/resolution/entities/bundle_entity.go

```diff
@@ -87,13 +88,20 @@ func (b *BundleEntity) PackageName() (string, error) {
 	return b.bundlePackage.PackageName, nil
 }
 
-func (b *BundleEntity) Version() (*bsemver.Version, error) {
+func (b *BundleEntity) VersionBlang() (*bsemver.Version, error) {
 	if err := b.loadPackage(); err != nil {
 		return nil, err
 	}
 	return b.semVersion, nil
 }
 
+func (b *BundleEntity) VersionMasterminds() (*mmsemver.Version, error) {
+	if err := b.loadPackage(); err != nil {
+		return nil, err
+	}
+	return mmsemver.NewVersion(b.bundlePackage.Version)
```

Comment:
Nit: the asymmetry of `VersionBlang()` stuff being parsed during `b.loadPackage()` and the masterminds version being parsed here every time is maybe a bit unexpected and feels a bit off?

Other potential options:
- add a new sibling field to `b.semVersion`, parse both during `loadPackage()`
- parse into a blang version and then just build a masterminds.Version struct (either here or in `loadPackage`) using the blang version fields (this might be nice because its one fewer error to handle)
- have this `Version()` method just return a string and expect that caller to do the parsing.

Reply:
Another option: do the (option 2) conversion from blang version to masterminds version in the `InMastermindsVersionRange()` function.

Reply:
TL;DR: this is what happens when you try to optimize code coverage!

1. I had created a new field, but all errors occurred during blang code processing (since it's just a simple version), and subsequently the masterminds error handling code was never executed, lowering code coverage. I could get rid of the error handling code, but that's bad form.
2. The `Masterminds.Version` struct has no user-accessible fields, so you can't create the structure that way. It actually makes more sense to do a `blang.Version.String()` and feed that into `Masterminds.NewVersion()`, but it's the same error handling problem as option 1.
3. This causes the version string to be parsed all the time, also non-optimal, and worse performance than option 1.

Reply:
> Another option: do the (option 2) conversion from blang version to masterminds version in the InMastermindsVersionRange() function.

AFAICT, this means passing an blang structure into a mastermind-named function, which seems bad form.

Reply:
> The Masterminds.Version struct has no user-accessible fields, so you can't create the structure that way.

I think you can do `masterminds.NewVersion(major, minor, patch, pre, metadata string)` and get back a pointer to the struct. It requires a bit of finagling because the blang pre and metadata fields are slices that need to be joined.

Reply:
But letting the caller do the parsing in option 3 seems fine to me too.

Reply:
You're right, blangToMasterminds() would basically be this: 
```
mmsemver.NewVersion(bundledVersion.String)
```
Which is just moving where the conversion takes place

Reply:
Sorry, I was thinking of this function: https://github.com/Masterminds/semver/blob/2f39fdc11c33c38e8b8b15b1f04334ba84e751f2/version.go#L214

Reply:
Although option 3 might be the cleanest, it is changing the behavior of `sortVersion()`.

Anything I do to try to change that _will_ negatively impact code coverage.

Reply:
IMO, trying to cover every error path just leads to brittle tests that make refactoring hard. I think as long as we're covering the happy paths, and the coverage "regression" is only happening because of some refactoring of when/how errors are handled, we shouldn't spend time dealing in the 1s or .1s of coverage percentages.

Reply:
Is there a way to specify a (minimum, err maximum? negative) threshold for CodeCov?

Reply:
>Sorry, I was thinking of this function: https://github.com/Masterminds/semver/blob/2f39fdc11c33c38e8b8b15b1f04334ba84e751f2/version.go#L214

Because of the pre and metadata fields, etc. it's just easier to do all conversions via `.String()`.